demo/App.scss

```diff
@@ -0,0 +1,26 @@
+#widgets-demo {
+  min-height: 100%;
+  display: flex;
+
+  .main {
+    display: flex;
+    flex: 1;
+  }
+
+  .content {
+    margin: 1rem;
+    width: 100%;
+  }
+
+  form {
+    padding: 1rem;
+  }
+
+  section {
+    background-color: white;
+    margin: 1rem auto 0;
+    padding: .5rem 1rem;
+    box-shadow: 0 .0625rem .125rem 0 #c6c7ca;
+    border-radius: .25rem;
+  }
+}
```

Comment:
So I think the reason you had `main` is because the sidebar would not grow when the content did. This goes to using un-semantic markup for the sake of styling as the `widget-demo` wrapper already exists. So my solution would be to make the content scrollable, instead of the whole page.
Here's my CSS (includes all the other suggestions on markup)
```CSS
#widgets-demo {
  display: flex;
  height: 100%;
  width: 100%;

  .content {
    width: 100%;
    height: 100%;
    padding: 1rem;

    overflow: auto;
  }

  form {
    margin: 1rem;
  }

  section {
    margin: 1rem 0;
    padding: 0.5rem 1rem;

    background-color: white;
    box-shadow: 0 0.0625rem 0.25rem 0 #C6C7CA;
    border-radius: 0.25rem;
  }
}